webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -76,28 +109,54 @@ const ModalForm: React.FC<Props> = ({ form, user, groups, viewOnly }) => {
         title: 'Name',
       },
       {
-        dataIndex: 'isGlobal',
-        key: 'isGlobal',
-        render: (val: boolean) => (val ? <Icon name="checkmark" /> : ''),
-        title: 'Global?',
-      },
-      {
-        dataIndex: 'workspaceOnly',
-        key: 'workspaceOnly',
-        render: (val: boolean) => (val ? <Icon name="checkmark" /> : ''),
-        title: 'Workspaces?',
+        dataIndex: 'permissions',
+        key: 'permissions',
+        render: (p) => p.length,
+        title: 'Permissions',
       },
-    ];
-    if (canModifyPermissions && !viewOnly) {
+    ] as ColumnDef<UserRole>[];
```

Comment:
Why are you casting here?

Reply:
This is to enforce a type on column object, to avoid unexpected field

Reply:
You're not actually _enforcing_ something when you do a type cast, you're _removing_ a guarantee. What's the unexpected field?

Reply:
I think what you want here is probably `const columns: ColumnDef<UserRole>[] = ...`

Reply:
Got it, thanks!